marketplace/src/app/components/open/product/product.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component } from '@angular/core';
+import {ActivatedRoute, Router} from "@angular/router";
+import staticRandomInt from "../../../helpers/staticRandomInt";
+import {ProductService} from "../../../service/product.service";
+import {MenuItem} from "primeng/api";
+
+@Component({
+  selector: 'app-add-product',
+  templateUrl: './product.component.html'
+})
+export class ProductComponent {
+  loading: boolean = true;
+
+  product: any;
+  images: any;
```

Comment:
images not in product?

Reply:
Check rule 34, the images are mapped to objects that the gallery component takes in